File: imtecho-ui/app/manage/smstype/services/sms-type.service.js
(function () {
    function SmsTypeService($resource, APP_CONFIG) {
        var api = $resource(APP_CONFIG.apiPath + '/smstype/:action/:subaction', {}, {
            getAllSmsTypes: {
                method: 'GET',
                isArray: true
            },
            updateSmsTypeState: {
                method: 'PUT',
                params: {
                    action: 'toggleState'
                }
            },
            createSmsType: {
                method: 'POST',
                params: {
                    action: 'create'
                }
            },
            updateSmsType: {
                method: 'PUT',
                params: {
                    action: 'updateSmsType'
                }
            },
            getSmsTypeByType: {
                method: 'GET'
            }
        });
        return {
            getAllSmsTypes: function () {
                return api.getAllSmsTypes({
                    action: 'getAllSmsTypes'
                }, {}).$promise;
            },
            updateSmsTypeState: function (smsType, isActive) {
                return api.updateSmsTypeState({is_active: isActive}, smsType).$promise
            },
            createSmsType: function (smsType) {
                return api.createSmsType({}, smsType).$promise;
            },
            updateSmsType: function (smsType) {
                return api.updateSmsType({}, smsType).$promise
            },
            getSmsTypeByType: function (type) {
                return api.getSmsTypeByType({
                    action: 'getSmsTypeByType',
                    type: type
                }, {}).$promise;
            }
        };
    }
    angular.module('imtecho.service').factory('SmsTypeService', ['$resource', 'APP_CONFIG', SmsTypeService]);
})();

File: imtecho-web/src/main/java/com/argusoft/imtecho/smstype/service/SmsTypeService.java
package com.argusoft.imtecho.smstype.service;

import com.argusoft.imtecho.smstype.dto.SmsTypeMasterDto;

import java.util.List;

/**
 * <p>
 * Define methods of sms type service
 * </p>
 *
 * @author monika
 * @since 10/03/21 12:20 PM
 */
public interface SmsTypeService {
//    List<SmsTypeMasterDto> getAllActiveSmsTypes();

    List<SmsTypeMasterDto> getAllSmsTypes();

    void toggleActive(SmsTypeMasterDto smsTypeMasterDto, Boolean isActive);

    void create(SmsTypeMasterDto smsTypeMasterDto);

    SmsTypeMasterDto getSmsTypeByType(String type);

    void updateSmsType(SmsTypeMasterDto smsTypeMasterDto);
}
